applications/annotate_points.md

```diff
@@ -0,0 +1,383 @@
+@viewer.bind_key('.')
+def next_label(event=None):
+    """Keybinding to advance to the next label with wraparound"""
+    
+    # get the currently selected label
+    current_properties = points_layer.current_properties
+    current_label = current_properties['label'][0]
+    
+    # determine the index of that label in the labels list
+    ind = list(labels).index(current_label)
+    
+    # increment the label with wraparound 
+    new_ind = (ind + 1) % len(labels)
+    
+    # get the new label and assign it
+    new_label = labels[new_ind]
+    current_properties['label'] = np.array([new_label])
+    points_layer.current_properties = current_properties
+```
+
+We can do the same with another function that instead decrements the label with wraparound.
+
+```python
+@viewer.bind_key(',')
+def prev_label(event):
+    """Keybinding to decrement to the previous label with wraparound"""
+    current_properties = points_layer.current_properties
+    current_label = current_properties['label'][0]
+    ind = list(labels).index(current_label)
+    n_labels = len(labels)
+    new_ind = ((ind - 1) + n_labels) % n_labels
+    new_label = labels[new_ind]
+    current_properties['label'] = np.array([new_label])
+    points_layer.current_properties = current_properties
+```
+
+## Mousebinding to iterate through labels
+Similar to keybindings, we can also bind functions to mouse events such as clicking or dragging. Here, we create a function that will increment the label after a point is added (i.e., the mouse is clicked in the viewer canvas when in the point adding mode). This is convenient for quickly adding all labels to a frame, as one can simply click each feature in order without having to manually swap labels. To achieve this, we first check if the points layer is the the adding mode (`layer.mode == 'add'`). If so, we then reuse the next_label() function we defined above in the keybindings to increment the label. Finally, 
+
+```python
+def next_on_click(layer, event):
+    """Mouse click binding to advance the label when a point is added"""
+    # only do something if we are adding points
+    if layer.mode == 'add':
+        next_label()
+
+        # by default, napari selects the point that was just added
+        # disable that behavior, as the highlight gets in the way
+        layer.selected_data = []
+```
+
+After creating the function, we then add it to the `points_layer` mouse drag callbacks. In napari, clicking and dragging events are both handled under the `mouse_drag_callbacks`. For more details on how mouse event callbacks work see the examples [[1](https://github.com/napari/napari/blob/e9e411491ce2e3758a1e9a5600fe4d317ed57088/examples/custom_mouse_functions.py), [2](https://github.com/napari/napari/blob/e452850854702ed4f37a826d20295ad54a6c510e/examples/mouse_drag_callback.py)].
+
+```python
+# bind the callback to the mouse drag event
+points_layer.mouse_drag_callbacks.append(next_on_click)
+```
+
+## Saving annotation keybinding
```

Comment:
I think this can now be done with our builtin csv writer on the points layer? If not would be good to know why and maybe adapt our builtin writer so it can meet the needs of this example. I'd much rather point people towards using the built in then having to add their own `Control-S` keybinding, which will overwrite ours.

Reply:
I guess this saving maybe is done with some special deeplabcut multindexing, which makes sense that it is custom, so maybe I'm asking too much for one example, but curious what you make of the idea.

Also curious if it is possible to add hookimpls in scripts like this? Maybe @tlambert03 knows? That would be pretty cool, because then you could add a custom `napari_write_points` hookimpl rather than having to add a custom `Control-S` keybinding

Reply:
We can definitely write the points using the builtin writer. The function here writes out the DLC format, which I think is a little different from the builtin writer format. I was thinking it would be cool to make a follow up application turning that function into a writer plugin. However, I am also cool with making this tutorial using the built-in and the a follow up one on making a writer plugin.

In either case, you make a good point about the usage of `Control-S`.

Reply:
Yeah, maybe let's have a dedicated application note making a DLC reader/ writer plugin - could even think then about contributing it to DLC. For now for this example let's just use the builtin

Reply:
Sounds good. I think using the built-in writer also helps keep the focus on using the core napari features.

Reply:
I think we can actually drop the whole concept of `output_path` from this example and just instruct people to do `Control-S` with the layer selected type in the path. We should have a section explaining this, but I don't think we need to have a custom keybinding. It might be nice to include a screenshot with the save dialog open overlaid on the viewer in this section too

Reply:
Done!